tracer/src/Datadog.Trace.Coverage.collector/FiltersHelper.cs

```diff
@@ -0,0 +1,158 @@
+
+        var regexes = AttributesRegexes.GetOrAdd(
+            filters,
+            list =>
+            {
+                var lstRegex = new List<Regex>(list.Count);
+                foreach (var item in list)
+                {
+                    lstRegex.Add(new Regex(item, RegexOptions.Compiled));
+                }
+
+                return lstRegex;
+            });
+
+        foreach (var regex in regexes)
+        {
+            if (regex.IsMatch(attributeFullName))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static bool FilteredBySourceFile(string sourcePath, IReadOnlyList<string> filters)
+    {
+        if (filters.Count == 0 || string.IsNullOrWhiteSpace(sourcePath))
+        {
+            return false;
+        }
+
+        var value = Matchers.GetOrAdd(
+            filters,
+            list =>
+            {
+                var instance = new Matcher();
+                var lstRegex = new List<Regex>();
+                foreach (var filter in list)
+                {
+                    if (filter is null)
+                    {
+                        continue;
+                    }
+
+                    if (!filter.Contains("**"))
+                    {
+                        try
+                        {
+                            lstRegex.Add(new Regex(filter, RegexOptions.Compiled));
+                        }
+                        catch
+                        {
+                            // .
+                        }
+                    }
+
+                    instance.AddInclude(Path.IsPathRooted(filter) ? filter.Substring(Path.GetPathRoot(filter).Length) : filter);
+                }
+
+                return Tuple.Create(instance, (IReadOnlyList<Regex>)lstRegex);
+            });
+
+        var matcher = value.Item1;
+        // https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.filesystemglobbing.matcher?view=dotnet-plat-ext-6.0
+        var globbingResult = matcher.Match(Path.IsPathRooted(sourcePath) ? sourcePath.Substring(Path.GetPathRoot(sourcePath).Length) : sourcePath).HasMatches;
+        if (globbingResult)
+        {
+            return true;
+        }
+
+        foreach (var regex in value.Item2)
+        {
+            if (regex.IsMatch(sourcePath))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static bool FilteredByAssemblyAndType(string module, string? type, IReadOnlyList<string> filters)
+    {
+        if (filters.Count == 0)
+        {
+            return false;
+        }
+
+        module = Path.GetFileNameWithoutExtension(module);
+        if (module == null)
+        {
+            return false;
+        }
+
+        foreach (var filter in filters)
+        {
+            var typePattern = filter.Substring(filter.IndexOf(']') + 1);
+            var modulePattern = filter.Substring(1, filter.IndexOf(']') - 1);
```

Comment:
Need to be careful about checking filter length here maybe? e.g. this will throw if filter = `""`.

I wonder if it's worth slapping a `try-catch` inside each of these `FilteredBy*` methods?

Reply:
changed in https://github.com/DataDog/dd-trace-dotnet/pull/4507/commits/f4f2b092e3ef51883207ae341c38312cc73c6b20